### PR TITLE
Pearson and Spearman pairwise correlation

### PR DESCRIPTION
**Pull request by rjenc29**
Pearson and Spearman pairwise correlation, using pluggable data scaling functions.  Enum of SKLearnDataSets for parameterised tests. 
All JIT compile; all significantly faster than equiv pandas; scipy; sklearn fns.

### CODE REVIEW

File: fastats/maths/correlation/pearson.py [hunk 1]
```diff
@@ -1,6 +1,8 @@
 
 import numpy as np
 
+from fastats.scaling.scaling import standard
```
Comment (pawroman) [inline fastats/maths/correlation/pearson.py]: I'd use the more concise form: `from fastats.scaling import standard`

File: fastats/scaling/scaling.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+from numpy import empty_like, mean, std, sqrt, argsort, ones, nonzero, empty
```
Comment (pawroman) [inline fastats/scaling/scaling.py:2]: We leave one blank line at the start and end of `.py` files by convention throughout the project

**Comment by pawroman** [comment]
Excellent work, all problems I see are purely cosmetic.

File: fastats/maths/correlation/pearson.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
...
+
 
 def pearson(x, y):
```
Comment (pawroman) [inline fastats/maths/correlation/pearson.py]: I wonder if these non-pairwise corrs can be expressed by pairwise? All we'd have to do is to construct a 2-D array ad-hoc

Comment (rjenc29) [reply fastats/maths/correlation/pearson.py]: That's possible, but there's a performance cost

File: fastats/scaling/scaling.py (hunk 2 above)
Comment (rjenc29) [reply fastats/scaling/scaling.py:2]: Good spot, will fix

**Comment by pawroman** [comment]
LGTM, thanks for committing these!

**Comment by rjenc29** [comment]
Appreciate the review and feedback - nice one, have a great weekend

**Pull request merged**